sentry_sdk/integrations/sanic.py

```diff
@@ -96,14 +96,23 @@ async def sentry_handle_request(self, request, *args, **kwargs):
 
         old_router_get = Router.get
 
-        def sentry_router_get(self, request):
-            # type: (Any, Request) -> Any
-            rv = old_router_get(self, request)
+        def sentry_router_get(self, *args):
+            # type: (Any, Union[Any, Request]) -> Any
+            rv = old_router_get(self, *args)
             hub = Hub.current
             if hub.get_integration(SanicIntegration) is not None:
                 with capture_internal_exceptions():
                     with hub.configure_scope() as scope:
-                        scope.transaction = rv[0].__name__
+                        if version >= (21, 3):
+                            # Sanic versions above and including 21.3 append the app name to the
+                            # route name, and so we need to remove it from Route name so the
+                            # transaction name is consistent across all versions
+                            sanic_app_name = self.ctx.app.name
+                            scope.transaction = rv[0].name.split(
+                                "%s." % sanic_app_name
+                            )[1]
```

Comment:
Could the name.split list length ever be 0 or 1?

Reply:
I do not think so https://github.com/sanic-org/sanic/blob/108a4a99c7e12561b35b62e26ecfc33c77db802d/sanic/mixins/routes.py#L666

Reply:
`self.name` cannot be null

Reply:
I will add safe guards regardless